src/plugins/data/common/timefilter/types.ts

```diff
@@ -25,4 +25,5 @@ export interface RefreshInterval {
 export interface TimeRange {
   from: string;
   to: string;
+  mode?: 'absolute' | 'relative';
```

Comment:
Why do we need this explicit distinguisher between absolute and relative?

Reply:
The code expects mode to be one of those two values.  I could type it as `string` but it really should be an enum.  If you are asking why we need this at all, I believe it's because from/to are treated differently.  If relative, from/to will be like `now` or `now-1y`, if absolute, from/to will be a specific date/time.  So they are parsed differently.